sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Listeners/ServiceBusListener.cs

```diff
@@ -179,19 +179,22 @@ public async Task StopAsync(CancellationToken cancellationToken)
                 // CloseAsync method stop new messages from being processed while allowing in-flight messages to be processed.
                 if (_singleDispatch)
                 {
-                    if (_isSessionsEnabled)
+                    if (_sessionMessageProcessor.IsValueCreated)
```

Comment:
This might be introducing a subtle race condition if the processor somehow is created after this check. It would still potentially get cleaned up in Dispose, but that would again be a race condition. Probably safest to just use the isSessionsEnabled check.